### PR TITLE
Skip loading in TaskSpec

### PR DESCRIPTION
**Pull request by fferegrino**
Implements logic to skip dag loading via a parameter

### CODE REVIEW

**Comment by edublancas** [comment]
I'll take a detailed look at this over the next 2-3 days, in the meantime, can you fix the lining issues?

```
./tests/spec/test_dagspec.py:942:1: F811 redefinition of unused 'test_spec_with_sourceless_functions' from line 920
```

**Comment by fferegrino** [comment]
Oh, snap, sure; thought I had run yapf.

**Comment by fferegrino** [comment]
Can't manage to reproduce this error locally using Python 3.8, will try tomorrow in 3.6 as the pipelines are failing in that version.

**Comment by edublancas** [comment]
thanks! I'll review this over the weekend

**Comment by edublancas** [comment]
I want to make a few changes, can you give me [write permissions to your fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)?

**Comment by fferegrino** [comment]
> I want to make a few changes, can you give me [write permissions to your PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)?

Yeah, it is enabled.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM continuumio/miniconda3:latest
+
+ARG VERSION=3.6
+
+RUN pip install invoke
+
+WORKDIR ploomber
```
Comment (fferegrino) [inline Dockerfile:7]: This file is not needed, will remove before creating the final PR

**Comment by edublancas** [comment]
I made a few changes. the most important one was to remove the conditional in `DAGSpec.to_dag()` - `ploomber scaffold`  does not need a `DAG` object, and works with `DAGspec`.

Let's now wait for the CI and if this works, we'll cross off the first item in the list from #332  :)

**Comment by edublancas** [comment]
pretty weird that I cannot reproduce the failing tests - they pass locally. I'll take a look later again. please remove the files that aren't needed and any other modifications that you want to make and I'll merge

**Comment by fferegrino** [comment]
Done. 

As a separate question: is there any preference between rebase or merge to bring upstream changes into branches?

**Comment by fferegrino** [comment]
woha, I thought I had got the isolation right.

**Comment by edublancas** [comment]
yeah these failing tests are super weird, I cannot reproduce them locally. I'll check them later but will merge this now :)

**Comment by edublancas** [comment]
we prefer to rebase

**Pull request merged**